src/plugins/lua/LuaEvent.cpp

```diff
@@ -51,12 +51,8 @@ void LuaEventProcessor::AddEvent(std::string_view name, std::string_view express
 void LuaEventProcessor::RemoveEvent(std::string_view name)
 {
 	RemoveEvents({ std::string(name) });
-	auto it = std::find_if(m_eventDefinitions.begin(), m_eventDefinitions.end(),
-		[&name](const std::unique_ptr<LuaEvent>& event) { return event->GetName() == name; });
-	if (it != m_eventDefinitions.end())
-	{
-		m_eventDefinitions.erase(it);
-	}
+	m_eventDefinitions.erase(std::remove_if(m_eventDefinitions.begin(), m_eventDefinitions.end(), 
+		[&name](const std::unique_ptr<LuaEvent>& event) { return event->GetName() == name; }));
```

Comment:
erase needs 2nd param `m_eventDefinitions.end()` or you're just erasing the first one.

Reply:
Yeah, always forget that. Moot now anyway, the better solution is to fix the bug that allowed the same event ID on multiple events (PR updated)